tests/spec/core/highlight-spec.js

```diff
@@ -74,6 +105,79 @@ describe("Core — Highlight", () => {
     };
     const doc = await makeRSDoc(ops);
     const pre = doc.getElementById("test");
-    expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
+    expect(pre.querySelectorAll("span[class~=hljs-]").length).toBe(0);
+  });
+
+  it("checks if <pre> content is warped in <code>", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `${makeDefaultBody()}
+        <section>
+          <pre class="js">
+            function foo() {
+              alert('foo');
+            }
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+    const pre = doc.querySelector("pre");
+    const code = pre.firstElementChild;
+    expect(code.localName).toBe("code");
+    expect(code.classList.contains("hljs")).toBeTruthy();
+    expect(code.classList.contains("js")).toBeTruthy();
+  });
+
+  it("asynchronously hightlights <code> elements inside <pre>", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `${makeDefaultBody()}
+        <section>
+          <pre class="example" id="first-pre">
+            <code class="js" id="test1">
+              function one(){}
+            </code>
+            this function(){} is not highlighted.
+            <code class="http" id="test2">
+              Header: Test1
+            </code>
+          </pre>
+          <pre id="second-pre">
+            second function(){} is not highlighted.
+            <code class="js">
+              function three(){}
+            </code>
+            <code>
+              function four(){}
+            </code>
+            <code class="http">
+              Header: Test5
+            </code>
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const firstPre = doc.getElementById("first-pre");
+    expect(firstPre.textContent).toContain(
+      "this function(){} is not highlighted"
```

Comment:
Wouldn't this be always true even after highlighting? Should we check `innerHTML` instead?

Reply:
yes, you are right.